### PR TITLE
test(core): add test coverage

### LINKED ISSUE #1: Cover project with specs
We need to cover this project with some specs, there're few classes to test.

### PR DESCRIPTION
**Pull request by rubynho**
# Test coverage

It closes #1 

### Chore
Add tests to coverage some aspects of the gem.

### CODE REVIEW

**Comment by boniattirodrigo** [comment]
The scenarios that you wrote are fine @rubynho. Just make sure that different actions don't block each other. The gem defines action and id as identifiers so if you call it twice with different actions or ids they can't block themselves.

Good job so far.

**Review by boniattirodrigo — APPROVED** [review]

File: spec/rate_limit_control_spec.rb [hunk 1]
```diff
@@ -3,7 +3,49 @@
...
+      id: 'foobar',
+      allowed_requests: 3,
+      storage: Redis.new,
+      timeout: 10
+    }
+  end
+
+  let(:action) { "Any action you would like to execute" }
+
+  context "when the actions" do
```
Comment (matheusazzi) [inline spec/rate_limit_control_spec.rb]: I think you don't need that `context "when the actions"`.
It's probably better having only:
```ruby
context "when actions exceed the total allowed"
context "when actions don't exceed the total allowed"
context "when there are actions being rate limited"
```

Comment (rubynho) [reply spec/rate_limit_control_spec.rb]: I usually do this way when I have two context blocks that start with the same phrase. But in this case I think it's better the way you supposed.

File: spec/rate_limit_control_spec.rb [hunk 2]
```diff
@@ -3,7 +3,47 @@
...
+
+  context "when an action stack is full" do
+    before do
+      expect_any_instance_of(described_class).not_to receive(:set_action_as_blocked)
+
+      3.times { described_class.new(action_config) { action } }
+    end
+
+    it "does not block other actions" do
+      action_config[:action] = 'rate_limit_example_2'
```
Comment (boniattirodrigo) [inline spec/rate_limit_control_spec.rb]: I think you could create a new let here with this different config or create a let just for the action name.

Like:
```
  let(:another_action_config) do
    {
      action: 'rate_limit_example_2',
      id: 'foobar',
      allowed_requests: 3,
      storage: Redis.new,
      timeout: 10
    }
  end
```

Or

```
  let(:action_name) { "rate_limit_example" }
  let(:another_action_config) do
    {
      action: action_name,
      id: 'foobar',
      allowed_requests: 3,
      storage: Redis.new,
      timeout: 10
    }
  end
```

And in this `it` you can create another `action_name`.

Comment (rubynho) [reply spec/rate_limit_control_spec.rb]: I like the idea!

**Pull request merged**